src/main/java/ch/njol/skript/expressions/ExprNumbers.java

```diff
@@ -81,10 +81,14 @@ public boolean init(final Expression<?>[] exprs, final int matchedPattern, final
 			s = f;
 			f = temp;
 		}
-		final double amount = integer ? Math.floor(f.doubleValue()) - Math.ceil(s.doubleValue()) + 1 : Math.floor(f.doubleValue() - s.doubleValue() + 1);
 		final List<Number> list = new ArrayList<>();
 		final double low = integer ? Math.ceil(s.doubleValue()) : s.doubleValue();
-		for (int i = 0; i < amount; i++) {
+		double decimalCheck = s.doubleValue() % 1;

```

Comment:
Your decimal source comes from the start value, so I think this code fails for something like 1 to 1.5

Reply:
Hmmm... Alright that's fine. In that case, please add another example for that case, and inform people that it'll the take the decimal precision of the first number and not the highest decimal precision of the two.